src/plugins/exitscript/exitscript.js

```diff
@@ -0,0 +1,166 @@
+* @author @ipaksc
+*/
+( function( $, window, wb ) {
+"use strict";
+var componentName = "wb-exitscript",
+	selector = "." + componentName,
+	initEvent = "wb-init" + selector,
+	$document = wb.doc,
+	defaults = {},
+	i18nDict = {
+		en: {
+			"exitscript-warning": "Warning",
+			"exitscript-exitMsg": "You are about to leave a secure site, do you wish to continue?",
+			"exitscript-yesMsg": "Yes",
+			"exitscript-cancelMsg": "Cancel"
+		},
+		fr: {
+			"exitscript-warning": "Avertissement",
+			"exitscript-exitMsg": "Vous êtes sur le point de quitter un site sécurisé. Voulez-vous continuer?",
+			"exitscript-yesMsg": "Oui",
+			"exitscript-cancelMsg": "Annuler"
+		}
+	},
+
+/**
+* @method init
+* @param {jQuery Event} event Event that triggered the function call
+*/
+	init = function( event ) {
+		var elm = wb.init( event, componentName, selector ),
+			$elm,
+			settings;
+		if ( elm ) {
+			$elm = $( elm );
+			settings = $.extend(
+		true,
+		defaults, {},
+		window[ componentName ],
+		wb.getData( $elm, componentName )
+	);
+			i18nDict = i18nDict[ wb.lang || "en" ];
+			i18nDict = {
+				warning: i18nDict[ "exitscript-warning" ],
+				exitMsg: i18nDict[ "exitscript-exitMsg" ],
+				yesMsg: i18nDict[ "exitscript-yesMsg" ],
+				cancelMsg: i18nDict[ "exitscript-cancelMsg" ]
+			};
+			
+			wb.ready( $elm, componentName );
+		}
+
+	};
+
+	// Add your plugin event handler
+	$document.on( "click", selector, function( ev ) {
+		ev.preventDefault();
+		var elm = event.currentTarget,
+				$elm = $( elm ),
+				moDalId,
+				wrapper,
+				exitURLsearchParam,
+				exitURL,
+				exitParam = encodeURIComponent( "targetUrl=" + String( this.href ) );
+		var moDal = document.createDocumentFragment();
+		var tpl = document.createElement( "div" );
+		var settings = $.extend(
+		defaults, {}
+		);
+
+		if(this.hasAttribute("target")){ 
+		  exitParam = exitParam + encodeURIComponent( "&targetBlank=true" );
+		}
+			 
+		exitURL = settings.exitURL;
+			
+		if( !exitURL.includes( "targetUrl" ) ){ 
+	
+		if( !wb.getUrlParts( exitURL ).search ){
+			exitURL = wb.getUrlParts( exitURL ).href + "?" + exitParam;
+			exitParam = exitParam.split( '%26' ).join( '&' );
+	
+		} else { 
+				exitURLsearchParam = String(wb.getUrlParts( exitURL ).search );
+				exitURLsearchParam = exitURLsearchParam.replace("?","");		
+				exitParam =  exitURLsearchParam + "&" + exitParam;
+				exitParam = exitParam.split('%26').join('&');
+				//Build exitURL
+				exitURL = wb.getUrlParts( exitURL ).protocol + 
+				"//" + 
+				wb.getUrlParts( exitURL ).host + 
+				wb.getUrlParts( exitURL ).pathname  + 
+				"?" + 
+				exitParam;
+			} 
+		}
+			var defaultModadisplay = settings.displayModal !== false;
+			
+			// Set default properties
+			if ( defaultModadisplay ) {
+			   
+				if(this.hasAttribute("target")){
+					settings.targetBlank = "target='_blank'"
+				} 
+				if ( !settings.exitMsg ) {
```

Comment:
Is those message is defined per link or it is a global setting?

Consider a instruction like this instead of the whole conditional block for one (1) instruction:
`var exitMsg = settings.exitMsg || i18nDict.exitMsg`